app/app.py

```diff
@@ -84,99 +88,141 @@ def get_intersect_features(query_geom, grid, idx):
 
-        if flow_key in grid["FeatureIndex"][intersect_cell_id]["properties"]:
+    params = { "$query" : query }
 
-            for trip_cell_id in grid["FeatureIndex"][intersect_cell_id]["properties"][flow_key].keys():
+    res = requests.get(TRIPS_URL, params)
 
-                count = grid["FeatureIndex"][intersect_cell_id]["properties"][flow_key][trip_cell_id][mode]
-                count_as_height =  count / 5    # each 5 trips will equate to 1 meter of height on the map
+    res.raise_for_status()
 
-                if count:
-                    if trip_cell_id not in trip_features_lookup:
-                        '''
-                        Add a new entry in the trip features lookup, dropping all feature
-                        properties except current count
-                        '''
-                        trip_features_lookup[trip_cell_id] = dict(grid["FeatureIndex"][trip_cell_id])
-                        trip_features_lookup[trip_cell_id]["properties"] = {
-                            "trips" : 0,
-                            "cell_id": int(trip_cell_id),
-                            "count_as_height": count_as_height
-                        }
+    return res.json()
 
-                    trip_features_lookup[trip_cell_id]["properties"]["trips"] += count                
-                    total_trips += count
 
-    # assemble matched features into geojson structure
-    trip_features = {"type": "FeatureCollection" }
+def build_geojson(grid, trips, flow_key_start):
+    '''
+    Combine trip counts with their corresponding geojson feature, returning a geojson
+    object with counts assigned to `trips` property
+    '''
+    geojson = {"type":"FeatureCollection","features":[]}
+
+    for cell in trips:
+        cell_id = cell.get(flow_key_start)
+        feature = grid.get(cell_id)
+
+        count = int(cell.get("trip_count"))
+
+        count_as_height =  count / 5  # each 5 trips will equate to 1 meter of height on the map
+
+        feature["properties"]["trips"] = count
+        feature["properties"]["count_as_height"] = count_as_height
+        feature["properties"]["cell_id"] = int(cell_id)
+        feature["properties"]["trips"] = count
+        geojson["features"].append(feature)
 
-    trip_features["features"] = [trip_features_lookup[cell_id] for cell_id in trip_features_lookup.keys()]
+    return geojson
 
-    return {"features": trip_features, "total_trips": total_trips}
 
+def get_total_trips(trips):
+    return sum([int(trip["trip_count"]) for trip in trips])
+    
 
 dirname = os.path.dirname(__file__)
-source = os.path.join(dirname, "data/grid.json")
+source = os.path.join(dirname, "data/hex500_indexed.json")
 
 with open(source, "r") as fin:
-    data = json.loads(fin.read())
-    idx = spatial_index(data["FeatureIndex"][feature_id] for feature_id in data["FeatureIndex"].keys())
+    # TRIPS_URL =  "https://data.austintexas.gov/resource/pqaf-uftu.json"
```

Comment:
this seems like code cruft, can we remove?

Reply:
☑️